### PR TITLE
Add Tables support

### LINKED ISSUE #14: DataFrames/Tables.jl support as inputs
Often one wants to convert an DataFrame/Table into Matrix.

DataFrames are naturally observation per row.
For a `AbstractDataFrame`  I think `Matrix(df)` makes it into a `Matrix`.
I am not sure if there is a generalization of this to all Tables?

### PR DESCRIPTION
**Pull request by oxinabox**
This closes #14 

An issue is, because Table is not a type can't dispatch on it.
so we wrap it in a `_TableHolder` for dispatch purposes.

One assumption this code makes is that a a table is never a AbstractMatrix.
Or rather that if it is, it should be treated using Matrix behavour, rather than Table behavour.
I think that is sound.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -10,16 +10,19 @@ Distributions = "31c24e10-a181-5473-b8eb-7969acd0382f"
...
 NamedDims = "356022a1-0364-5f58-8944-0da4b18d706f"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+Tables = "bd369af6-aec1-5ad0-b16a-f7cc5008161c"
 
 [compat]
 AxisArrays = "0.4"
 Compat = "3.3"
 Distributions = "0.21, 0.22"
 NamedDims = "0.2"
+Tables = "1"
```
Comment (oxinabox) [inline Project.toml:20]: Need v1 as there is a bug in v0.2.11 that has been fix now

That means this requires DataFrames v0.20 also

I think that is fine

File: src/ObservationDims.jl [hunk 2]
```diff
@@ -123,6 +141,25 @@ end
 # Any -> SingleObs: never any need to rearrage
 organise_obs(::SingleObs, data; obsdim=nothing) = data
 
+# Tabhles support
```
Comment (nicoleepp) [inline src/ObservationDims.jl]:
```suggestion
# Tables support
```

File: test/runtests.jl [hunk 3]
```diff
@@ -32,7 +33,7 @@ using Test
 
     @testset "organise_obs" begin
 
-        c_organise_obs(args...; kwargs...) = collect(organise_obs(args...; kwargs...))
+        corganise_obs(args...; kwargs...) = collect(organise_obs(args...; kwargs...))
```
Comment (nicoleepp) [inline test/runtests.jl]: Why this change? It seemed more readable before

Comment (oxinabox) [reply test/runtests.jl]: Good catch it was a bad find and replace on my part

**Review by nicoleepp — APPROVED** [review]
